### DISCUSSION TITLE
generate_appcast script?

### DISCUSSION THREAD

**Discussion by marioaguzman** [sparkle-project/Sparkle#2338] [Q&A] [answered]
After updating via SPM to Sparkle 2.3.2, I noticed that the script generate_appcast is no longer in the bin folder, which I had scripted to use that location. After looking around, I can't seem to find it but it seems there's a new generate_appcast Swift CLI for it? Is there documentation on how to build/use it? Unless I'm looking in the wrong place, current docs generally mention generate_appcast.

Thanks!

**Comment by zorgiepoo** [comment, marked as answer]
This hasn't changed.

https://sparkle-project.org/documentation/

If you're using SPM pay attention to:

> From Xcode’s project navigator, if you right click and show the Sparkle package in Finder, you will find Sparkle’s tools to generate and sign updates in ../artifacts/Sparkle/

Make sure you're looking in the `artifacts` directory.

Alternatively you can download the tools yourself from the [Sparkle releases](https://github.com/sparkle-project/Sparkle/releases) page. SPM makes it a bit awkward to access them.

**Reply by dgrcode** [reply]
Somehow I also missed that paragraph from the docs. I guess the title "Add the Sparkle framework to your project" made me follow the instructions until it was added to my project, which for me was just this part:
> If you use [Swift Package Manager](https://swift.org/package-manager/):
>
>  - In your Xcode project: File › Add Packages…
>  - Enter https://github.com/sparkle-project/Sparkle as the package repository URL

And probably never read after that.

Maybe a change like the following could help?
```diff
If you use [Swift Package Manager](https://swift.org/package-manager/):

+ You can find Sparkle's tools to generate and sign updates by going to Xcode’s project navigator,
+ right click and show the Sparkle package in Finder. The tools are inside ../artifacts/sparkle/Sparkle/bin/
+ (in Finder you may need to go up one folder from checkouts via Go › Enclosing Folder).

 - In your Xcode project: File › Add Packages…
 - Enter https://github.com/sparkle-project/Sparkle as the package repository URL
 - Choose the Package Options. The default options will let Xcode automatically update versions of Sparkle 2.

- From Xcode’s project navigator, if you right click and show the Sparkle package in Finder, you will find
- Sparkle’s tools to generate and sign updates in ../artifacts/sparkle/Sparkle/bin/ (in Finder you may need
- to go up one folder from checkouts via Go › Enclosing Folder).
```

I've moved the paragraph before the instructions, and started with "You can find Sparkle's tools...", so the reader can identify the paragraph is relevant in the first few words.

To be honest, if it wasn't for this discussion I would still going around trying to find where to find the /bin folder.

**Reply by zorgiepoo** [reply]
It is logical for the instructions to using SPM precede the note about where to find the tools. One thing we could do is indent the paragraph however (if possible) to make it more clear it is still relevant to SPM. Same treatment could be done for the Carthage section too.

**Reply by dgrcode** [reply]
I agree my proposal is not the optimal solution. I just think it would have been better for me to have it that way, but that doesn't necessarily mean that it would be better for most people.

Having a second look at the docs, how would you feel about having 1.x sections like this?
```text
1. Add the Sparkle framework to your project
  1.1. Using Swift Package Manager
  ...

  1.2. Using CocoaPods:
  ... 
```

That way you can easily see the instructions are actually quite short and easy to read vs a long text.

Probably this combined with indenting the paragraph makes it easier for people to find?

**Reply by zorgiepoo** [reply]
Maybe, not sure how it will look. If you want to play around with it, the website repository is here and you can open a PR https://github.com/sparkle-project/sparkle-project.github.io

**Reply by dgrcode** [reply]
Awesome, I'll give it a go and open a PR

**Reply by dgrcode** [reply]
just opened https://github.com/sparkle-project/sparkle-project.github.io/pull/207. Feel free to close it if you don't like the result.
